### ISSUE TITLE
Not always working on windows

### ISSUE THREAD

**Issue by dgallion1** [getlantern/systray#149]
Windows
This version works in all cases: v0.0.0-20190727060347-6f0e5a3c556c
Newer versions work on my dev machine but not always on other machines.
No errors, icon appears in systray
No menu is shown when clicking.
Event viewer:
Microsoft-Windows-DistributedCOM     EventID 10016 

That seems to be associated with my application launch.

**Comment by hick** [comment]
yes, I got icon in systray bug no menu also,  bu just on windows 10 , and it seems like  it's ok on windows 7

**Comment by CsterKuroi** [comment]
It's related to #148

**Comment by xiaobingcaicai** [comment]
i get the same problem, win10 and win7 usually

**Comment by xiaobingcaicai** [comment]
> yes, I got icon in systray bug no menu also, bu just on windows 10 , and it seems like it's ok on windows 7

after some minutes? maybe hours?

**Comment by xiaobingcaicai** [comment]
> Windows
> This version works in all cases: v0.0.0-20190727060347-6f0e5a3c556c
> Newer versions work on my dev machine but not always on other machines.
> No errors, icon appears in systray
> No menu is shown when clicking.
> Event viewer:
> Microsoft-Windows-DistributedCOM EventID 10016
> 
> That seems to be associated with my application launch.

have you solved?

**Comment by Doraemonkeys** [comment]
I also encountered the same problem. After a period of time, the menu of the app did not display. Have you solved it?

**Comment by OlliSchm** [comment]
> I also encountered the same problem. After a period of time, the menu of the app did not display. Have you solved it?

I've got the same issue on some clients running an application with systray.

**Comment by Doraemonkeys** [comment]
> > I also encountered the same problem. After a period of time, the menu of the app did not display. Have you solved it?
> 
> I've got the same issue on some clients running an application with systray.

I have solved this problem as detailed in the code below.

```go
func main() {
	runtime.UnlockOSThread()

	go func() {
		runtime.LockOSThread() // systray.Run must be called on a locked OS thread
		systray.Run(onReady, nil)
	}()

	// do something else
}
```

**Comment by OlliSchm** [comment]
> > > I also encountered the same problem. After a period of time, the menu of the app did not display. Have you solved it?
> > 
> > 
> > I've got the same issue on some clients running an application with systray.
> 
> I have solved this problem as detailed in the code below.
> 
> ```go
> func main() {
> 	runtime.UnlockOSThread()
> 
> 	go func() {
> 		runtime.LockOSThread() // systray.Run must be called on a locked OS thread
> 		systray.Run(onReady, nil)
> 	}()
> 
> 	// do something else
> }
> ```

Hey, thanks a lot, I'll give it a try. I'm also running the systray in a go-routine, so it seems like it might fix that for me as well.
